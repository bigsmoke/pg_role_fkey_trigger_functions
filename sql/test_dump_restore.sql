\o /dev/null
select not :{?test_stage} as test_stage_missing, not :{?extension_name} as extension_name_missing;
\o
\gset
\if :test_stage_missing
    \warn 'Missing `:test_stage` variable.'
    \quit
\endif
\if :extension_name_missing
    \warn 'Missing `:extension_name` variable.'
    \quit
\endif
\o /dev/null
select :'test_stage' = 'pre-dump' as in_pre_dump_stage;
\o
\gset

\set SHOW_CONTEXT 'errors'

\if :in_pre_dump_stage
    create extension pg_role_fkey_trigger_functions with cascade;
\endif

call test_dump_restore__maintain_referenced_role(:'test_stage'::text);
