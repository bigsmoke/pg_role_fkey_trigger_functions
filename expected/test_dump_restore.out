-- createdb
-- psql -X -f 'sql/test_dump_restore.sql' -v 'extension_name=pg_role_fkey_trigger_functions' -v 'test_stage=pre-dump'
CREATE EXTENSION
CALL
-- pg_dumpall --roles-only
-- pg_dump --format=custom --file <dump_file>
-- kill -9 <pg_pid>
-- rm -r $PGDATA
-- psql postgres -X -c '\set ON_ERROR_STOP' -f <roles_dump_file>
SET
SET
SET
CREATE ROLE
ALTER ROLE
CREATE ROLE
ALTER ROLE
CREATE ROLE
ALTER ROLE
ALTER ROLE
ALTER ROLE
ALTER ROLE
GRANT ROLE
-- createdb 'test_dump_restore_oid_noise'
-- psql -X -f 'sql/test_dump_restore.sql' -v 'extension_name=pg_role_fkey_trigger_functions' -v 'test_stage=pre-restore'
DO
-- pg_restore --create --dbname postgres <dump_file>
-- psql -X -f 'sql/test_dump_restore.sql' -v 'extension_name=pg_role_fkey_trigger_functions' -v 'test_stage=post-restore'
CALL
